Class App\\AnonymousClass1c24b64c37740a293d326ceb80ab1552 was not found while trying to analyse it

# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.9.18              |
| Installed as   | prefixed                           |

I'm trying to run the PHP 7.0 set and getting a very weird error:
```
Analyze error: \"Class App\\AnonymousClass1c24b64c37740a293d326ceb80ab1552 was not found while trying to analyse it - discovering symbols is probably not configured properly.\". Include your files in \"$parameters->set(Option::AUTOLOAD_PATHS, [...]);\
```

that is caused by this code (that I shrunk down to bare minimum to reproduce it):
```php
<?php

declare(strict_types=1);

namespace App;

use Exception;

final class Bug
{
    public function weird() : void
    {
        $exception = new class() extends Exception {
            public static function create() : self
            {
                $exception = new self();

                return $exception;
            }
        };
    }
}
```

Important to reproduce: you must be in a namespace + use anonymous class.

I have no idea what causes this error. I ran it locally with `--debug` and it didn't give me anything valuable:
```
vendor/bin/rector process src/Bug.php --debug
Rector 0.9.18
Config file: rector.php

[parsing] src/Bug.php
[refactoring] src/Bug.php
    [applying] Rector\Renaming\Rector\Name\RenameClassRector
    [applying] Rector\Renaming\Rector\FileWithoutNamespace\PseudoNamespaceToNamespaceRector
    [applying] Rector\PHPUnit\Rector\Class_\RemoveDataProviderTestPrefixRector
    [applying] Rector\PHPUnit\Rector\Class_\TestListenerToHooksRector
    [applying] Rector\PHPUnit\Rector\Class_\AddProphecyTraitRector
    [applying] Rector\Php56\Rector\FunctionLike\AddDefaultValueForUndefinedVariableRector
    [applying] Rector\Php70\Rector\ClassMethod\Php4ConstructorRector
    [applying] Rector\Php70\Rector\FunctionLike\ExceptionHandlerTypehintRector
    [applying] Rector\Renaming\Rector\MethodCall\RenameMethodRector
    [applying] Rector\PHPUnit\Rector\ClassMethod\ExceptionAnnotationRector
    [applying] Rector\Renaming\Rector\ClassMethod\RenameAnnotationRector
    [applying] Rector\TypeDeclaration\Rector\ClassMethod\AddParamTypeDeclarationRector
    [applying] Rector\TypeDeclaration\Rector\ClassMethod\AddReturnTypeDeclarationRector
    [applying] Rector\MysqlToMysqli\Rector\Assign\MysqlAssignToMysqliRector
    [applying] Rector\Php70\Rector\Assign\ListSplitStringRector
    [applying] Rector\Php70\Rector\Assign\ListSwapArrayOrderRector
    [applying] Rector\Php53\Rector\Variable\ReplaceHttpServerVarsByServerRector
    [applying] Rector\Php70\Rector\Variable\WrapVariableVariableNameInCurlyBracesRector
[post rectors] src/Bug.php
[printing] src/Bug.php



 [ERROR] Could not process "src/Bug.php" file, due to:
         "Analyze error: "Class App\AnonymousClass21dfecc387e6b0ae3223745c946461c7 was not found while trying to
         analyse it - discovering symbols is probably not configured properly.". Include your files in
         "$parameters->set(Option::AUTOLOAD_PATHS, [...]);" in "rector.php" config.
         See https://github.com/rectorphp/rector#configuration".

```

If you can point me in the right direction I can try and fix it via a PR. 

## Minimal PHP Code Causing Issue

https://getrector.org/demo/9d0ad78d-3b89-4f60-bfc1-9b3172a1d715
## Expected Behaviour

No error.

--- a/packages/post-rector/src/Rector/NameImportingPostRector.php
+++ b/packages/post-rector/src/Rector/NameImportingPostRector.php
@@ -83,6 +83,10 @@ final class NameImportingPostRector extends AbstractPostRector
 
     private function processNodeName(Name $name): ?Node
     {
+        if ($name->isSpecialClassName()) {
+            return $name;
+        }
+
         $importName = $this->getName($name);
 
         if (! is_callable($importName)) {

--- a/rules/code-quality/tests/Rector/Return_/SimplifyUselessVariableRector/Fixture/special_class_name_with_anonymous_class.php.inc
+++ b/rules/code-quality/tests/Rector/Return_/SimplifyUselessVariableRector/Fixture/special_class_name_with_anonymous_class.php.inc
@@ -1,0 +1,45 @@
+<?php
+
+namespace Rector\CodeQuality\Tests\Rector\Return_\SimplifyUselessVariableRector\Fixture;
+
+use Exception;
+
+final class SpecialClassNameWithAnonymousClass
+{
+    public function weird() : void
+    {
+        $exception = new class() extends Exception {
+            public static function create() : self
+            {
+                $exception = new self();
+
+                return $exception;
+            }
+        };
+
+        throw $exception;
+    }
+}
+?>
+-----
+<?php
+
+namespace Rector\CodeQuality\Tests\Rector\Return_\SimplifyUselessVariableRector\Fixture;
+
+use Exception;
+
+final class SpecialClassNameWithAnonymousClass
+{
+    public function weird() : void
+    {
+        $exception = new class() extends Exception {
+            public static function create() : self
+            {
+                return new self();
+            }
+        };
+
+        throw $exception;
+    }
+}
+?>

--- a/rules/code-quality/tests/Rector/Return_/SimplifyUselessVariableRector/SimplifyUselessVariableRectorTest.php
+++ b/rules/code-quality/tests/Rector/Return_/SimplifyUselessVariableRector/SimplifyUselessVariableRectorTest.php
@@ -6,6 +6,7 @@ namespace Rector\CodeQuality\Tests\Rector\Return_\SimplifyUselessVariableRector;
 
 use Iterator;
 use Rector\CodeQuality\Rector\Return_\SimplifyUselessVariableRector;
+use Rector\Core\Configuration\Option;
 use Rector\Testing\PHPUnit\AbstractRectorTestCase;
 use Symplify\SmartFileSystem\SmartFileInfo;
 
@@ -21,6 +22,8 @@ final class SimplifyUselessVariableRectorTest extends AbstractRectorTestCase
      */
     public function test(SmartFileInfo $fileInfo): void
     {
+        $this->setParameter(Option::AUTO_IMPORT_NAMES, true);
+
         $this->doTestFileInfo($fileInfo);
     }
 

